src/xopen/__init__.py

```diff
@@ -196,12 +196,19 @@ def __init__(self, path, mode='r', threads=None):
         """
         if mode not in ('r', 'rt', 'rb'):
             raise ValueError("Mode is '{0}', but it must be 'r', 'rt' or 'rb'".format(mode))
-        if threads is None:
-            threads = min(_available_cpu_count(), 4)
-        
+
         pigz_args = ['pigz', '-cd', path]
-        if threads != 0:
+        if threads:  # Check if threads is not 0 or None.
             pigz_args += ['-p', str(threads)]
+        else:
+            # Single threaded behaviour by default because:
+            # - Using a single thread to read a file is the least unexpected
+            #   behaviour. (For users of xopen, who do not know which backend is used.)
+            # - There is quite a substantial overhead (+25% cpu time) when
+            #   using multiple threads while there is only a 10% gain in wall
+            #   clock time. This is a bad trade off.
```

Comment:
Nitpicking: Please remove "This is a bad trade off." (The reader can decide that for themselves.) Please write uppercase "CPU".